pkg/reactor/reactor.go

```diff
@@ -165,55 +169,189 @@ func printResolved(ctx context.Context, manifest *api.Documentation, writer io.W
 // - NodeSelectors
 // The resulting model is the actual flight plan for replicating resources.
 func (r *Reactor) resolveStructure(ctx context.Context, nodes []*api.Node, globalLinksConfig *api.Links) error {
-	var handler resourcehandlers.ResourceHandler
 	for _, node := range nodes {
 		node.SetParentsDownwards()
 		if len(node.Source) > 0 {
-			if handler = r.ResourceHandlers.Get(node.Source); handler == nil {
-				return fmt.Errorf("No suitable handler registered for URL %s", node.Source)
-			}
-			if len(node.Name) == 0 {
-				node.Name = "$name"
+			nodeName, err := r.resolveNodeName(ctx, node)
+			if err != nil {
+				return err
 			}
-			name, ext := handler.ResourceName(node.Source)
-			id := uuid.New().String()
-			node.Name = strings.ReplaceAll(node.Name, "$name", name)
-			node.Name = strings.ReplaceAll(node.Name, "$uuid", id)
-			node.Name = strings.ReplaceAll(node.Name, "$ext", fmt.Sprintf(".%s", ext))
+			node.Name = nodeName
 		}
 		if node.NodeSelector != nil {
-			if handler = r.ResourceHandlers.Get(node.NodeSelector.Path); handler == nil {
-				return fmt.Errorf("No suitable handler registered for path %s", node.NodeSelector.Path)
-			}
-			if err := handler.ResolveNodeSelector(ctx, node, node.NodeSelector.ExcludePaths, node.NodeSelector.FrontMatter, node.NodeSelector.ExcludeFrontMatter, node.NodeSelector.Depth); err != nil {
+			newNode, err := r.resolveNodeSelector(ctx, node, globalLinksConfig)
+			if err != nil {
 				return err
 			}
-			// remove node selectors after resolution
+			node.Nodes = append(node.Nodes, newNode.Nodes...)
```

Comment:
Also consider that with the current GitHub implementation for `ResolveNodeSelector`, if the URL is not resolved to a github `ResourceLocator` it will return `nil` error and `nil` node, and on this line you will get NPE.